library/general/uniqueingresshost/template.yaml

```diff
@@ -5,6 +5,24 @@ metadata:
   annotations:
     metadata.gatekeeper.sh/title: "Unique Ingress Host"
     metadata.gatekeeper.sh/version: 1.0.0
+    metadata.gatekeeper.sh/requiresSyncData: |
+      "[
+        {
+          "group":"extensions",
```

Comment:
Do all K8s versions have all of these GVKs? Interesting to think about cases where there may be missing GVKs on older clusters (or newer clusters if the "extensions" group has been removed)

Reply:
Ah interesting. Maybe this schema needs a slight tweak to account for this, eg:
requiresSyncData: [list of {requirement}s, ANDed]
{requirement}: [list of {GVK}s, ORed]

That would give enough expression to account for things like "v1beta1 fully deprecated in favor of v1" or "Deployments migrated from extensions to apps"

Reply:
I wouldn't presume kind to be stable, but otherwise +1

Reply:
Thanks @maxsmythe @ekitson !  Based on the feedback the PR has been updated to use this [ [{}OR{}] AND [{}OR{}] ] format:

```
    metadata.gatekeeper.sh/requiresSyncData: |
      "[
        [
          {
            "groups":["extensions"],
            "versions": ["v1beta1"],
            "kinds": ["Ingress"]
          },
          {
            "group": ["networking.k8s.io"],
            "version": ["v1beta1", "v1"],
            "kind": ["Ingress"]
          }
        ]
      ]"
```

Reply:
Both the schema and detection method have been updated.